prospector/filter_rank/rules.py

```diff
@@ -89,12 +89,15 @@ def apply_rule_changes_relevant_path(
     candidate: CommitWithFeatures, advisory_record: AdvisoryRecord
 ):
     rule_tag = "CH_REL_PATH"
-    if candidate.changes_relevant_path:
-        explanation_template = "This commit touches the following relevant paths: '{}'"
-        relevant_paths = [
-            changed_path in advisory_record.paths
-            for changed_path in candidate.commit.changed_files
```

Comment:
This is just an educated guess.
The `candidate.commit.changed_files` are not set (maybe it was set before in the constructor or some other already refactored parts), so it is a `[]`.
![image](https://user-images.githubusercontent.com/2602591/126450287-a8db8908-e93d-4d6b-bdb0-abd54d2ff25d.png)

Reply:
I made a major refactoring of the rules.py module, could you take a look there too?
(I wanted to make a PR to discuss, but I was in a hurry...)